src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediateStatement.java

```diff
@@ -77,6 +86,36 @@ private static ResultType extractResultType(Statement statement) throws SQLExcep
     }
   }
 
+  // Split statements by ';' delimiter, but ignore anything that is nested with '' or "".
+  private List<String> splitStatements(String sql) {
+    List<String> statements = new ArrayList<>();
+    boolean quoteEsacpe = false;
+    int index = 0;
+    for (int i = 0; i < sql.length(); ++i) {
+      if (sql.charAt(i) == SINGLE_QUOTE) {
+        quoteEsacpe = !quoteEsacpe;
+      }
+      if (sql.charAt(i) == STATEMENT_DELIMITER && !quoteEsacpe) {
+        statements.add(sql.substring(index, i + 1).trim());
+        index = i + 1;
+      }
+    }
+
+    if (index < sql.length()) {
+      statements.add(sql.substring(index, sql.length()).trim());
+    }
+    return statements;
+  }
+
+  protected List<String> parseStatements(String sql) {
+    Preconditions.checkNotNull(sql);
+    List<String> statements = splitStatements(sql);
+    if (statements.size() > 1) {
+      batchedQuery = true;
```

Comment:
Nit: Instead of defining a new property, I'd probably use statements.size() > 1 to all places used. It's much clearer what it means.

Reply:
Done.